packages/salesforce-adapter/src/client/client.ts

```diff
@@ -256,6 +228,33 @@ export default class SalesforceClient {
       || realConnection(credentials.isSandbox, retryOptions || DEFAULT_RETRY_OPTS)
   }
 
+  private static saveOrgId(_err: Error, userInfo: UserInfo): void {
+  // Note that this returns the 18 char organization ID and not the 15 you see in the website.
+    SalesforceClient.recentOrgId = userInfo.organizationId
+  }
+
+  public static async getRemainingDailyRequests(creds: Credentials): Promise<number> {
+    const conn = realConnection(creds.isSandbox, {
+      maxAttempts: 2,
+      retryStrategy: RetryStrategies.HTTPOrNetworkError,
+    })
+    await conn.login(creds.username, creds.password + (creds.apiToken ?? ''), SalesforceClient.saveOrgId)
```

Comment:
I'll ask again because github marks things as outdated when they move and people tend to ignore them:
why use context and side effects instead of a return value?
why can't you use the connection like all the other functions do and just get the return value?
```suggestion
    const { organizationId } = await conn.login(creds.username, creds.password + (creds.apiToken ?? ''))
```

Reply:
Sorry i have a problem with the notifications didn't see your comment.  I'll fix this in the next commit. In general the call back isn't our API it's jsforce's api. I fixed it so the context is in the scope of the function though. Which i think should be more elegant.

Reply:
look at all the other usages of jsforce.
it returns a value

Reply:
Cool didn't notice, thanks. Fixed